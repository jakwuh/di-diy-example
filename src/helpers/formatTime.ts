import {addHours, format} from 'date-fns';

export const offsetToGMT = (offset: number) => {
    let sign = offset >= 0 ? '+' : '-';
    let absOffset = Math.abs(offset);
    let zero = absOffset > 9 ? '' : '0';

    return sign + zero + absOffset + ':00';
};


const getUTCDate = (date) => {
    return new Date(
        date.getUTCFullYear(),
        date.getUTCMonth(),
        date.getUTCDate(),
        date.getUTCHours(),
        date.getUTCMinutes(),
        date.getUTCSeconds(),
    );
};

export const offsetCurrentTimeFactory = (offset: number) => (currentTime: Date) => {
    const utc = getUTCDate(currentTime);

    return format(addHours(utc, offset), 'DD MMM, YYYY HH:mm:ss');
};
