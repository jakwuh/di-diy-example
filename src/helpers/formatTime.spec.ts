import {offsetCurrentTimeFactory, offsetToGMT} from './formatTime';

test('offsetToGMT', () => {
    expect(offsetToGMT(0)).toBe('+00:00');
    expect(offsetToGMT(-1)).toBe('-01:00');
    expect(offsetToGMT(-12)).toBe('-12:00');
    expect(offsetToGMT(10)).toBe('+10:00');
});

test('offsetCurrentTimeFactory', () => {
    const offsetTime = offsetCurrentTimeFactory(2);
    const date = new Date(2017, 4, 3, 4, 53, 44);

    expect(offsetTime(date)).toMatchSnapshot();
});
