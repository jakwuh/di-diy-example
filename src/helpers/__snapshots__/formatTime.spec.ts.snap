// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offsetCurrentTimeFactory 1`] = `"03 May, 2017 03:53:44"`;
